### PR TITLE
Add TasteDive Command

### LINKED ISSUE #856: TasteDive Command
> TasteDive is a recommendation engine that helps you explore your taste! Create an account for a better discovery experience. Once signed in, you receive better, personalized recommendations for similar music, movies, TV shows, books, authors, games, and podcasts, based on what you like.

https://tastedive.com/read/api

### PR DESCRIPTION
**Pull request by ghost**
Adds a tastedive command 

Closes #856

Edit: Idk why my last commit came with this pull request. I had pulled request from other branch.

### CODE REVIEW

**Comment by AsianCat54x** [comment]
@Doges, Oops! I guess I left out *e* everywhere. I will fix this, my bad!

**Review by fernando457829 — DISMISSED** [review]

**Review by Doges — DISMISSED** [review]

**Review by almeidx — APPROVED** [review]
Besides https://github.com/SwitchbladeBot/switchblade/pull/1346#discussion_r735595083, LGTM

File: src/commands/misc/tastedive.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  constructor (client) {
+    super({
+      name: 'tastedive',
+      requirements: { apis: ['tastedive'] },
+      parameters: [
+        {
+          type: 'string',
+          full: false,
+          whitelist: MEDIA_WHITE_LIST,
+          missingError: 'commands:searchplate.notFound'
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:15]: Wrong command

File: src/locales/en-US/commands.json [hunk 2]
```diff
@@ -672,6 +672,16 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "tastedive": {
+    "commandDescription": "Provides similar types of media according to user's liking",
+    "commandUsage": "<media> <liking>",
+    "place": "TasteDive",
+    "title": "Showing Results for {{title}}.",
+    "noMedia": "You will have to give me a valid type of media",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:680]: This is not being used

```suggestion
```

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -672,6 +672,16 @@
...
+    "availableMedia": "All the available media's are",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:681]: This is not being used

```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/misc/tastedive.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+      requirements: { apis: ['tastedive'] },
+      parameters: [
+        {
+          type: 'string',
+          full: false,
+          whitelist: MEDIA_WHITE_LIST,
+          missingError: 'commands:tastedive.notFound' 
+        }, {
+          type: 'string',
+          full: true,
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:18]: Probably would want to make this field required and use the `noText` message

File: src/locales/en-US/commands.json [hunk 5]
```diff
@@ -672,6 +672,14 @@
     "commandUsage": "<text>",
     "noText": "You have to give me some text to asciify!"
   },
+  "tastedive": {
+    "commandDescription": "Provides similar types of media according to user's liking",
+    "commandUsage": "<media> <liking>",
+    "place": "TasteDive",
```
Comment (almeidx) [inline src/locales/en-US/commands.json:678]:
```suggestion
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/misc/tastedive.js [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+          required: true,
+          missingError: 'commands:tastedive.notFound'
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:20]:
```suggestion
          missingError: 'commands:tastedive.noText' 
```

File: src/commands/misc/tastedive.js [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+  constructor (client) {
+    super({
+      name: 'tastedive',
+      requirements: { apis: ['tastedive'] },
+      parameters: [
+        {
+          type: 'string',
+          full: false,
+          whitelist: MEDIA_WHITE_LIST,
+          missingError: 'commands:tastedive.notFound'
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:15]:
```suggestion
          missingError: 'commands:tastedive.noText' 
```

**Review by almeidx — CHANGES_REQUESTED** [review]

File: src/commands/misc/tastedive.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+          full: true,
+          required: true,
+          missingError: 'commands:tastedive.noText' 
+        }
+      ]
+    }, client)
+  }
+
+  async run ({ channel, t, message }, media, term) {
+    const data = await this.client.apis.tastedive.search(media, term)
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:27]: We'd probably want to send an error message to the user if nothing is found

Comment (AsianCat54x) [reply src/commands/misc/tastedive.js:27]: I did that, but @Doges said that it's fine if I just log the error in the console

Comment (almeidx) [reply src/commands/misc/tastedive.js:27]: I'll leave that up to him then

Comment (Doges) [reply src/commands/misc/tastedive.js:27]: > 
> 
> I did that, but @Doges said that it's fine if I just log the error in the console

Nah, I said that we have a CommandError system

Comment (Doges) [reply src/commands/misc/tastedive.js:27]: https://github.com/SwitchbladeBot/switchblade/blob/5b08e03088605d1756a807cbcee975083a87dec3/src/commands/games/minecraft/skin.js#L28

Comment (AsianCat54x) [reply src/commands/misc/tastedive.js:27]: So should I send an embed saying the error?

Comment (Doges) [reply src/commands/misc/tastedive.js:27]: No, the command error already does that, when it called it sends and error with the string, example
image
https://github.com/SwitchbladeBot/switchblade/blob/5b08e03088605d1756a807cbcee975083a87dec3/src/commands/games/minecraft/server.js#L36

Comment (AsianCat54x) [reply src/commands/misc/tastedive.js:27]: Oh I see!

**Review by almeidx — DISMISSED** [review]

**Comment by almeidx** [comment]
CI is failing

**Review by almeidx — DISMISSED** [review]

**Comment by AsianCat54x** [comment]
So this command has no more issues, right?

**Review by Doges — DISMISSED** [review]
LGTM

**Comment by AsianCat54x** [comment]
Fixed the conflict here too :)

**Review by almeidx — APPROVED** [review]

File: src/commands/misc/tastedive.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  }
+
+  parseResponse (t, title, data) {
+    const description = data.map((item, index) => `\`${String(index).padStart(2, '0')}\`:  *${item.Name}*`).join('\n')
+
+    const embed = new SwitchbladeEmbed()
+      .setTitle(t('commands:tastedive.title', { title }))
+      .setDescription(description)
+
+    return embed
```
Comment (almeidx) [inline src/commands/misc/tastedive.js:43]:
```suggestion
    return new SwitchbladeEmbed()
      .setTitle(t('commands:tastedive.title', { title }))
      .setDescription(description)
```

**Review by almeidx — CHANGES_REQUESTED** [review]

**Review by almeidx — APPROVED** [review]

**Pull request closed without merging**